### PR TITLE
Move autograd detach to loss metrics

### PR DESCRIPTION
**Pull request by cbalioglu**
A small PR that handles autograd detaching of loss tensors within metric bags to be on the safe side.

### CODE REVIEW

File: src/fairseq2/nn/utils/gradient.py [hunk 1]
```diff
@@ -75,7 +75,7 @@ def forward(ctx: Any, x: Tensor, scale: float) -> Tensor:  # type: ignore[overri
 
         ctx.scale = scale
 
-        return x.detach().clone().requires_grad_(True)
```
Comment (kauterry) [inline src/fairseq2/nn/utils/gradient.py:78]: Why do we detach and clone here? Why not just return x? This removes x from the autograd graph right, why is it necessary?

File: src/fairseq2/recipes/wav2vec2/asr/train.py [hunk 2]
```diff
@@ -511,7 +511,7 @@ def __call__(self, batch: Seq2SeqBatch) -> tuple[Tensor, int]:
 
         loss = output.compute_loss(batch.target_seqs, batch.target_padding_mask)
 
-        self._metric_bag.update_ctc_loss(batch, loss.detach())
```
Comment (kauterry) [inline src/fairseq2/recipes/wav2vec2/asr/train.py:514]: Why do we do this? I don't understand.
